plugins/bitbucket-cloud-common/src/BitbucketCloud.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetch, { Request, Response } from 'node-fetch';
+import { BitbucketCloudConfig } from './BitbucketCloudConfig';
+import {
+  PaginatedRepositories,
+  Repository,
+  SearchCodeSearchResult,
+  SearchResultPage,
+} from './models';
+import { WithPagination } from './pagination';
+import {
+  FilterAndSortOptions,
+  PartialResponseOptions,
+  RequestOptions,
+} from './types';
+
+/** @public */
+export class BitbucketCloud {
```

Comment:
Maybe append Client or something to mark its use? We often have FooApi interface + FooClient implementation pairs

Reply:
adjusted